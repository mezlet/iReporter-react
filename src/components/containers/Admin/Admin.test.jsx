import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import ConnectedAdminPage, { AdminPage } from './AdminPage';

const props = {
  updateStatus: {
    success: false
  },
  incident: {
    data: {},
    isLoading: true
  },
  viewRecords: jest.fn(),
  statusUpdate: jest.fn(),
  auth: {
    isLoggedIn: true
  }
};

const initialState = {
  auth: {
    isLoggedIn: false,
    incident: {},
    success: false,
    status: {},
    errors: {},
    message: ''
  }
};

const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

const connectedWrapper = shallow(
  <Provider store={store}>
    <Router>
      <ConnectedAdminPage {...props} />
    </Router>
  </Provider>
);

const wrapper = shallow(<AdminPage {...props} />);

describe('<AdminPage/> rendering', () => {
  it('should render without crashing', () => {
    expect(connectedWrapper).toMatchSnapshot();
  });

  it('should update records on incident status update', () => {
    wrapper.setProps({ updateStatus: { success: true } });
    expect(wrapper.instance().props.viewRecords).toHaveBeenCalled();
  });

  it('should update incident with \'rejected\' status', () => {
    wrapper.setProps({
      incident: {
        data: [
          {
            id: '1',
            comment: 'hhshdhhd',
            title: 'DBJABD'
          }
        ]
      }
    });
    const updateIncidentButton = wrapper.find('DropdownItem').at(0);

    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    updateIncidentButton.simulate('click');
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalled();
  });

  it('should update incident with \'resolved\' status', () => {
    wrapper.setProps({
      incident: {
        data: [
          {
            id: '1',
            comment: 'hhshdhhd',
            title: 'DBJABD'
          }
        ]
      }
    });
    const updateIncidentButton = wrapper.find('DropdownItem').at(1);

    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    updateIncidentButton.simulate('click');
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalled();
  });

  it('should update incident with \'under-investigation\' status', () => {
    wrapper.setProps({
      incident: {
        data: [
          {
            id: '1',
            comment: 'hhshdhhd',
            title: 'DBJABD'
          }
        ]
      }
    });
    const updateIncidentButton = wrapper.find('DropdownItem').at(2);

    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    updateIncidentButton.simulate('click');
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalled();
  });
});
