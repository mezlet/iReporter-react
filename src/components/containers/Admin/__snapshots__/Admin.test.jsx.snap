// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPage/> rendering should render without crashing 1`] = `ShallowWrapper {}`;
